drf_rw_serializers/generics.py

```diff
@@ -25,6 +30,42 @@ def get_serializer_class(self):
 
         return self.serializer_class
 
+    def get_serializer_class(self):
+        """
+        Return the class to use for the serializer.
+        Defaults to using `self.serializer_class`.
+        If the request method is GET, it tries to use `self.read_serializer_class`.
+        If the request method is not GET, it tries to use `self.write_serializer_class`.
+        If the specific serializer class for the request method is not set, it falls back to
+        `self.serializer_class`.
+        You may want to override this if you need to provide different
+        serializations depending on the incoming request.
+        (Eg. admins get full serialization, others get basic serialization)
+        """
+        if hasattr(self, "request"):
+            if self.request.method in ["GET"]:
+                assert (
+                    getattr(self, "read_serializer_class", None) is not None
+                    or self.serializer_class is not None
+                ), (
+                    "'%s' should either include a `read_serializer_class` or `serializer_class` "
+                    "attribute, or override the `get_read_serializer_class()` or "
+                    "`get_serializer_class()` method." % self.__class__.__name__
+                )
+                return self.get_read_serializer_class()
+            else:
+                assert (
+                    getattr(self, "write_serializer_class", None) is not None
+                    or self.serializer_class is not None
+                ), (
+                    "'%s' should either include a `write_serializer_class` or `serializer_class` "
+                    "attribute, or override the `get_write_serializer_class()` or "
+                    "`get_serializer_class()` method." % self.__class__.__name__
+                )
+                return self.get_write_serializer_class()
+
+        return self._get_serializer_class()
```

Comment:
Did you check if we got 100% branch-coverage here from tests?

Reply:
Yes, I did, we got the 100%

Reply:
[Uploading Screencast from 27-05-2024 10:41:03.webm…]()